src/bootstrap/src/core/config/config.rs

```diff
@@ -410,6 +410,217 @@ impl Config {
         flags: Flags,
         get_toml: impl Fn(&Path) -> Result<TomlConfig, toml::de::Error>,
     ) -> Config {
+        let change_id;
+        let bypass_bootstrap_lock;
+        let mut ccache = None;
+        let mut ninja_in_file = true;
+        let mut submodules = None;
+        let mut compiler_docs = false;
+        let mut library_docs_private_items = false;
+        let mut docs_minification = true;
+        let mut docs = true;
+        let mut locked_deps = false;
+        let vendor;
+        let mut target_config = HashMap::new();
+        let mut full_bootstrap = false;
+        let mut bootstrap_cache_path = None;
+        let mut extended = false;
+        let tools;
+        let mut tool = HashMap::new();
+        let mut sanitizers = false;
+        let mut profiler = false;
+        let omit_git_hash;
+        let skip;
+        let include_default_paths;
+        let rustc_error_format;
+        let json_output;
+        let compile_time_deps;
+        let mut test_compare_mode = false;
+        let color;
+        let mut patch_binaries_for_nix = None;
+        let stage0_metadata;
+        let android_ndk;
+        let optimized_compiler_builtins;
+
+        let stdout_is_tty = std::io::stdout().is_terminal();
+        let stderr_is_tty = std::io::stderr().is_terminal();
+
+        let on_fail;
+        let explicit_stage_from_cli;
+        let explicit_stage_from_config;
+        let stage;
+        let keep_stage;
+        let keep_stage_std;
+        let mut src = {
```

Comment:
Could you please group the variables that have an "interesting" (i.e. non-Default) default together? Although I assume that once you remove the `let XXX` vars and move them to the Config constructor, this will be less of an issue.